cEP-0018.md

```diff
@@ -0,0 +1,162 @@
+# Integration of ANTLR into coala core
+
+| Metadata     |                                         |
+| ------------ |-----------------------------------------|
+| cEP          | 0018                                    |
+| Version      | 0.1.0                                   |
+| Title        | Integration of ANTLR into coala core    |
+| Authors      | Viresh Gupta <viresh16118@iiitd.ac.in>  |
+| Status       | Proposed                                |
+| Type         | Feature                                 |
+
+# Abstract
+
+This document describes how an API based on ANTLR will be constructed and
+maintained.
+
+# Introduction
+
+ANTLR provides parsers for various language grammars and thus we can provide
+an interface from within coala and make it available to bear writers so that
+they can write advanced linting bears. This will aim at supporting the more
+flexible visitor based method of AST traversal (as opposed to listener based
+mechanisms).
+
+The proposal is to introduce a parallel concept to the coala-bears library,
+which will be called the coala-ast hereon.
+
+# Proposed Change
+
+Here is the detailed implementation stepwise:
+
+    1. There will be a separate repository named as coala-ast which will be
+    installable via ```pip install coala-ast```.
+    2. A new interface would be introduced in coala-core (```coala/coalib```),
```

Comment:
I don't understand what you mean by a new interface.

Reply:
By a new interface, I meant a new package that could be imported in order to use the ```ASTWalker``` and ```ASTLoader``` classes
Will change interface -> package to avoid confusion :sweat_smile: